app/Models/Extensions/Thumb.php

```diff
@@ -79,24 +79,38 @@ public static function createFromQueryable(Relation|Builder $photoQueryable, Sor
 
 	/**
 	 * Creates a thumbnail from the given photo.
+	 * On Livewire it will use by default small and small2x if available, thumb and thumb2x if not.
+	 * On Legacy it will use thumb and thumb2x.
 	 *
 	 * @param Photo|null $photo the photo
 	 *
 	 * @return Thumb|null the created thumbnail or null if null has been passed
 	 */
 	public static function createFromPhoto(?Photo $photo): ?Thumb
 	{
-		$thumb = config('app.livewire', true) === true
-			? $photo?->size_variants->getSmall() ?? $photo?->size_variants->getThumb()
+		$thumb = (config('app.livewire', true) === true && $photo?->size_variants->getSmall() !== null)
+			? $photo->size_variants->getSmall()
 			: $photo?->size_variants->getThumb();
 		if ($thumb === null) {
 			return null;
 		}
 
-		$thumb2x = config('app.livewire', true) === true
-			? $photo?->size_variants->getSmall2x() // We do note provide '?? sv->Thumb2x()' because if messes up on retina display
+		$thumb2x = (config('app.livewire', true) === true && $photo?->size_variants->getSmall() !== null)
+			? $photo->size_variants->getSmall2x()
 			: $photo?->size_variants->getThumb2x();
 
+		/**
+		 * TODO: Code for later when Livewire is the only front-end.
+		 */
+		// $thumb = $photo?->size_variants->getSmall() ?? $photo?->size_variants->getThumb();
+		// if ($thumb === null) {
+		// 	return null;
+		// }
```

Comment:
Yeah, this is... more subtle than it may seem.

From what I remember, (square) thumbs are _guaranteed_ to be generated -- if necessary, through upscaling. small/small2x are not if the original is of a particularly low resolution.

Though even that "guaranteed" above is actually not always true -- thumbs might be missing for video files and I think also for password-protected albums. So the frontend (do we still call it so?) most likely already has code to handle such cases correctly.